VideoLocker/src/main/java/org/edx/mobile/view/CourseUnitNavigationActivity.java

```diff
@@ -291,16 +291,17 @@ public Fragment getItem(int pos) {
             //FIXME - for the video, let's ignore studentViewMultiDevice for now
             if (unit instanceof VideoBlockModel &&
                     ((VideoBlockModel) unit).getData().encodedVideos.getPreferredVideoInfo() != null) {
-                    unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
+                unitFragment = CourseUnitVideoFragment.newInstance((VideoBlockModel) unit);
             } else if (unit instanceof VideoBlockModel &&
                     ((VideoBlockModel) unit).getData().encodedVideos.getYoutubeVideoInfo() != null) {
-                    unitFragment = CourseUnitOnlyOnYoutubeFragment.newInstance(unit);
+                unitFragment = CourseUnitOnlyOnYoutubeFragment.newInstance(unit);
+            } else if (unit.getType() == BlockType.DISCUSSION) {
+                unitFragment = CourseUnitDiscussionFragment.newInstance(unit, courseData);
             } else if (!unit.isMultiDevice()) {
                 unitFragment = CourseUnitMobileNotSupportedFragment.newInstance(unit);
             } else if (unit.getType() != BlockType.VIDEO &&
                     unit.getType() != BlockType.HTML &&
                     unit.getType() != BlockType.OTHERS &&
-                    unit.getType() != BlockType.DISCUSSION &&
```

Comment:
This should be reinstated after implementing my suggestion at https://github.com/edx/edx-app-android/pull/724/commits/e7be548b08da1052dc54c431f54ffdf408f61515#r65494516